pkg/proxy/ipvs/proxier.go

```diff
@@ -2006,6 +2020,16 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
+		if len(newEndpoints) == 0 {
+			if onlyNodeLocalEndpoints {
```

Comment:
Incrementing internal or external should be based on the traffic policy instead of nodeLocalEndpoints. Also, the feature gate check should only happen for internal traffic policy. Something like:

```
if onlyNodeLocalEndpoints && len(newEndpoints) == 0 {
    if svcInfo.NodeLocalInternal() &&  utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
        proxier.serviceNoEndpointsTotalInternal++
    }

    if svcInfo.NodeLocalExternal() {
        proxier.serviceNoEndpointsTotalExternal++
    }
}
```

Reply:
onlyNodeLocalEndpoints is false when the external traffic policy is set. I'll wait for Dan's PR #106497 to merge before fixing this as it will give me better access at querying the traffic policy directly.